examples/research_projects/codeparrot/scripts/codeparrot_training.py

```diff
@@ -140,7 +137,10 @@ def evaluate(args):
         losses.append(accelerator.gather(loss))
         if args.max_eval_steps > 0 and step >= args.max_eval_steps:
             break
-    loss = torch.mean(torch.cat(losses))
+    losses = torch.cat(losses)
+    eval_size = eval_dataloader.dataset.current_size
+    losses = losses[:eval_size]
+    loss = torch.mean(losses)
```

Comment:
Maybe we could wrap that into a single line:
```suggestion
    loss = losses[:eval_dataloader.dataset.current_size].mean()
```